### DISCUSSION TITLE
Categorize after Object Detect

### DISCUSSION THREAD

**Discussion by Xtr3amS** [ultralytics/ultralytics#18237] [Q&A]
Hello,

We can detect objects in photos using YOLO. So can we categorize these results?

I am trying to do the following: For example, I have a folder with 15 thousand photos. After scanning the photos in this folder with YOLO and performing object detection, I want to select them according to the objects it finds. For example, I want to be able to select photos containing cats, seats, and people. Can I tag them or, for example, have them put photos containing cats in one folder and photos containing people in a different folder? Or can I find photos containing seats with any command? I hope I could explain what I mean.

To summarize, I want to have 15000 photos detected with yolo and then search and select them according to the objects they contain.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Xtr3amS, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for general guidance and examples of YOLO operations. Many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples are available there, and they might help you get started with your use case.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) including code, dependencies, and data to allow us to reproduce and assist more effectively.

For custom ❓ Questions like this one, it helps to include relevant details such as your code implementation, examples of your data, and logs or outputs you’ve observed. If you're working with object detection results and categorization, it might also be helpful to verify that you're following our guides, such as the [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/).

Considering your interest in categorizing and organizing photos based on YOLO’s object detection outputs, this could likely involve post-processing of YOLO results in Python. YOLO outputs include bounding boxes and class labels, which you should be able to use to achieve the desired functionality. Ultralytics engineers will assist further soon! 😊

## Upgrade

Ensure you are running the latest version of the `ultralytics` package in a [**Python>=3.8**](https://www.python.org/) environment:

```bash
pip install -U ultralytics
```

## Environments

For fast setup and experimentation, YOLO can be used in any of the following verified environments (complete with dependencies preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

For additional guidance or to share your results, join our vibrant Ultralytics community:

- Real-time discussions: [Discord](https://discord.com/invite/ultralytics) 🎧
- Community Q&A: [Discourse](https://community.ultralytics.com)
- Engaging discussions: [Subreddit](https://reddit.com/r/Ultralytics)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the CI badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, meaning that YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) have been verified across macOS, Windows, and Ubuntu.

**Comment by glenn-jocher** [comment]
@Xtr3amS yes, you can achieve this by running YOLO for object detection, extracting the predictions (e.g., class labels and bounding box data), and then writing a script to group or move images based on detected classes. The results object in YOLO provides the necessary data for this task. For implementation details, refer to the [Ultralytics YOLOv8 documentation on result extraction](https://docs.ultralytics.com/modes/predict/). Let us know if you need further clarification!
